src/test/java/io/gingersnapproject/cdc/MultiCacheManagerTest.java

```diff
@@ -106,28 +111,38 @@ public void testLeaveAndReturn() throws Exception {
          assertTrue(identifier.uri().equals(hotRodURI) || identifier.uri().equals(additionalMember));
       }
 
+      eventually(() -> "Engine not restarted!", () -> {
+         var engine = engines.get(additionalId);
+         ManagedEngine.Status status = Utils.extractField(engine, "status");
+         return status == ManagedEngine.Status.RUNNING;
+      }, 10, TimeUnit.SECONDS);
       eventually(() -> "Additional member engine not started: " + additionalId, () -> {
          try {
-            verify(notification, times(1)).connectorStarted(eq(additionalId), any());
+            verify(notification, times(2)).connectorStarted(eq(additionalId), any());
             return true;
          } catch (Throwable ignore) {
             return false;
          }
-      }, 5, TimeUnit.SECONDS);
-      eventually(() -> "Engine not restarted!", () -> {
-         var engine = engines.get(additionalId);
-         ManagedEngine.Status status = Utils.extractField(engine, "status");
-         return status == ManagedEngine.Status.RUNNING;
-      }, 15, TimeUnit.SECONDS);
-      verify(notification, never()).connectorFailed(any(), any());
+      }, 10, TimeUnit.SECONDS);
+
+      if (!Profiles.isProfileActive("oracle")) verify(notification, never()).connectorFailed(any(), any());
```

Comment:
I'll add a few more checks here for the `oracle` profile. The reason here is the fact Debezium is unable to start engines concurrently with an Oracle connector. The engine tries to create a flush table, and there is a concurrency issue that results in the engine failing: https://github.com/debezium/debezium/blob/f8c0e477e6a0aeffad004e8d48d5042843f87655/debezium-connector-oracle/src/main/java/io/debezium/connector/oracle/logminer/logwriter/CommitLogWriterFlushStrategy.java#L99-L101

In real usage, this will cause the engine to close, enter into retries, and eventually connect. I'll add more checks for this scenario, which is specific to oracle.